sale_last_price_info/models/product.py

```diff
@@ -8,13 +8,16 @@ class ProductProduct(models.Model):
 
     def _compute_last_sale(self):
         """ Get last sale price, last sale date and last customer """
-        lines = self.env['sale.order.line'].search(
-            [('product_id', '=', self.id),
-             ('state', 'in', ['sale', 'done'])]).sorted(
-            key=lambda l: l.order_id.date_order, reverse=True)
-        self.last_sale_date = lines[:1].order_id.date_order
-        self.last_sale_price = lines[:1].price_unit
-        self.last_customer_id = lines[:1].order_id.partner_id
+        so_line_obj = self.env['sale.order.line']
+        for product in self:
+            lines = so_line_obj.search(
+                [('product_id', '=', product.id),
+                 ('state', 'in', ['sale', 'done'])]).sorted(
+                key=lambda l: l.order_id.date_order, reverse=True)
```

Comment:
That's what I thought but I decided to not change (Minimal diff policy) the behaviour (I know this doesn't change any functional behaviour)

However, I will add that field and Improve `_compute_last_sale` accordingly if we are getting same reviews from other @oca member too.

Reply:
I fully agree with the remark of @sebalix, in the beginning you won't notice the performance issue.
But when using the system for a couple of time, it will get slower and slower.
Maybe a new module should be added, a module that adds the date_order as related stored field on the sale order line. Then this can be used to solve this suggested change?

Reply:
No need for the stored field if you limit search to 1 and put the order on the `search` method directly (not after the results).

Reply:
@pedrobaeza  Do you mean like this?
`search( [('product_id', '=', product.id), ('state', 'in', ['sale', 'done'])], limit=1).sorted( key=lambda l: l.order_id.date_order, reverse=True)`

Reply:
@pedrobaeza this is not possible, because the field is only available on sale.order.
The search method does not work with order_id.date_order
`    def _check_qorder(self, word):
        if not regex_order.match(word):
            raise UserError(_('Invalid "order" specified. A valid "order" specification is a comma-separated list of valid field names (optionally followed by asc/desc for the direction)'))
        return True
`
To be able to use it in the search, it should be a database field defined on the sale.order.line itself.

Reply:
I think we should go ahead with adding a related field in this module itself.

Reply:
You have 2 options then:

1. Add the date as related stored field in `sale.order.line`. This is convenient that the SQL column is created and populated in the pre-init-hook through an SQL for low installation time. And then do what I have proposed before. It's recommended also to put a new that can't conflict with other fields, like `date_order_sale_last_price_info` (I know, a bit cumbersome but it's the only way to assure uniqueness`, similar to what is done on onchanges).
2. Use the SQL view model `sale.report`, doing the search there, which is going to be more performant any way than current solution.

Reply:
> @Swapnesh-SerpentCS IMO best will be to search on sale.order with order='date_order desc' and limit=1 with product_id in the domain.

This would not be possible as we also need to get `price_unit` from related order line. (We have to use additional search for `sale.order.line where id=%s and product_id=%s` to get Order line if we use this way)